modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala

```diff
@@ -260,12 +281,28 @@ object Sequence {
 
       override def getCurrentBreakpoint: Boolean = false
 
+      override def setOperator(name: String): State = operatorL.set(self, Some(name))
+
+      override def setObserver(name: String): State = observerL.set(self, Some(name))
+
       override val done: List[Step[Result]] = seq.steps
 
       override def mark(i: Int)(r: Result): State = self
 
       override val toSequence: Sequence[Action \/ Result] = seq.map(_.right)
 
+      private val sequenceL: Final @> Sequence[Result] =
```

Comment:
`Final` is the case class representing the final state of a `Sequence.State`. It is used when the Zipper is advancing. However this case class was not introduced in this PR, I just needed a Lens for it.

I guess this is confusing when reading the diffs but I think it's more clear when considering the whole file because `Final` is scoped inside the `Sequence.State`.